firebase-firestore/src/main/java/com/google/firebase/firestore/local/QueryEngine.java

```diff
@@ -87,7 +97,39 @@ public ImmutableSortedMap<DocumentKey, Document> getDocumentsMatchingQuery(
       return result;
     }
 
-    return executeFullCollectionScan(query);
+    QueryContext context = new QueryContext();
+    result = executeFullCollectionScan(query, context);
+    if (result != null && automaticIndexingEnabled) {
+      CreateCacheIndexes(query, context, result.size());
```

Comment:
nit: The method name `CreateCacheIndexes` should start with a lowercase letter, conventionally.